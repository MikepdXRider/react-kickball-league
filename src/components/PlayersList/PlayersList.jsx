import React from 'react';
import PropTypes from 'prop-types';
import PlayerCard from '../PlayerCard/PlayerCard.jsx';
import { Link } from 'react-router-dom';


export default function PlayersList({playersDataArr}) {
    return (
        <ul>
            {
                playersDataArr.map(playerDataObj => {
                    return (
                        <Link to={`/players/${playerDataObj.id}`}>
                            <li key={playerDataObj.id}>
                                <PlayerCard playerDataObj={playerDataObj} />
                            </li>
                        </Link>
                    )
                })
            }
        </ul>
    )
};

PlayersList.propTypes = {
    playersDataArr: PropTypes.arrayOf(PropTypes.object)
};