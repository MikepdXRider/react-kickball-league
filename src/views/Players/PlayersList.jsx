import React from 'react';
import { Link } from 'react-router-dom';
import { useEffect, useState } from 'react/cjs/react.development';
import PlayerCard from '../../components/PlayerCard/PlayerCard.jsx';
import { getPlayers } from '../../services/players.jsx';

export default function PlayersList() {
    
    const [playersDataArr, setPlayersDataArr] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        async function getPlayersData() {
            const newPlayersDataArr = await getPlayers();
            // console.log('getPlayers fn returns: ', newPlayersDataArr);
            setPlayersDataArr(newPlayersDataArr);
        }

        getPlayersData();
        setIsLoading(false);
    }, []);
   
    return (
        <main>
            {
                isLoading
                ? <h1>Loading...</h1> 
                : <section>
                    <ul>
                        {playersDataArr.map(playerDataObj => {
                            return (
                                <li key={playerDataObj.id}>
                                    <Link to={`/players/${playerDataObj.id}`}>
                                        <PlayerCard playerDataObj={playerDataObj}/>
                                    </Link>
                                </li>
                            )
                        })}
                    </ul>
                </section>
            }
        </main>
    )
}
