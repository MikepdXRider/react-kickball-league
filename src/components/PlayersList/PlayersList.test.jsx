import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import PlayersList from "./PlayersList.jsx";

//need to mock out the server for this to work again. 
const playersDataArr = [{
    created_at: 'test-utc',
    id: 1,
    name: 'test-name',
    team_id: 3
}]

it.skip('renders PlayersList component', () => {
    const { container } = render(
    <MemoryRouter>
        <PlayersList playersDataArr={playersDataArr} /> 
    </MemoryRouter>
    )
    const playerName = screen.getByText(playersDataArr[0].name);

    expect(playerName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})