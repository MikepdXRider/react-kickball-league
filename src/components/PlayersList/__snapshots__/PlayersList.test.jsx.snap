// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayersList component 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/players/1"
      >
        <article>
          <h3>
            test-name
          </h3>
          <h6>
            Date Created: 
            Invalid Date
          </h6>
          <h5>
            Position: 
          </h5>
        </article>
      </a>
    </li>
  </ul>
</div>
`;
